unity/Assets/Scripts/DroneFPSAgentController.cs

```diff
@@ -88,6 +98,10 @@ void FixedUpdate()
             //here, the in-editor axisAlignedBoundingBox metadata for drone objects seems to be offset by some number of updates.
             //it's unclear whether this is only an in-editor debug draw issue, or the actual metadata for the axis
             //aligned box is messed up, but yeah.
+
+            if (this.agentState == AgentState.PendingFixedUpdate) {
+                this.agentState = AgentState.ActionComplete;
```

Comment:
not very familiar with the dron controller, but semantically should this be done at the end of FixedUpdate?

Reply:
At first I didn't follow what you meant, but I get it now.  Since the state transition indicates that we are no longer in PendingFixedUpdate and FixedUpdate only finishes at the end of the method, this transition should reside there.  I moved the block.  I don't think this changes the behavior, but I agree with your comment.